### ISSUE TITLE
WRC 10 FIA World Rally Championship (1462810)

### ISSUE THREAD

**Issue by whizse** [ValveSoftware/Proton#5143]
# Compatibility Report
- Name of the game with compatibility issues: WRC 10 FIA World Rally Championship
- Steam AppID of the game: 1462810

## System Information
- GPU: AMD Radeon RX 570
- Driver/LLVM version: Mesa 21.2.0/LLVM 12.0.1
- Kernel version: 5.10.0
- Link to full system information report as [Gist](https://gist.github.com/whizse/21c0812faa3767bee3d0e77a779edcdf):
- Proton version: Proton Experimental (experimental-6.3-20210830)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 

The game starts and seem to work but have similar issues to WRC 8 (tracked in bug #4567), it needs Proton Experimental to run. It will also need xinput1_3 set to native, or Windows version to 7.

The game still crashes because of problems with the DX12 renderer. [Bug report with trace here](https://github.com/HansKristian-Work/vkd3d-proton/issues/789).

As a workaround, the game can be configured to use the DX11 renderer. This needs be set in a configuration file, but the game crashes before the file is created. I have saved a [working copy (set to DX11) here.](https://gist.github.com/whizse/ee6686147382cc2a07756f1e9ac66037)

Place UserSettings.cfg in `<Steam-folder>/steamapps/compatdata/1462810/pfx/drive_c/users/steamuser/My Documents/My Games/WRC10`

My racing wheel (Logitech G29) is detected and works with force feedback, but only if Steam Input is disabled. I've seen reports of this happening on Windows too so it's probably not an issue unique to Proton. 

Extended gameplay have not been tried, network play have not been tested. There might be other issues. I noticed some intermittent problems loading the leaderboards but it might not at all be related to Proton.

**Comment by ivyl** [comment]
The game should be playable out of the box with today's experimental.

Steering wheels should work, but some other controllers may not, as they requires as of yet unimplemented windows.gaming.input.dll. This can be worked around by setting the game's prefix to win7 and overriding xinput1_3.dll with the builtin.

**Comment by xpander69** [comment]
Works great with Logitech G920.. FFB works fine also. Thank you for the proton-experimental update!
i'm playing with DX11 mode on my 1080Ti cause DX12 performance is half of the DX11 for me. 

Gameplay video with G920 Cam: https://youtu.be/8c8y61dN9w8

**Comment by xpander69** [comment]
There was a game update (3GB) recently and it seems i lost some of the ingame sounds for some reason. Co-driver is quiet, road effects like gravel stones hitting the car are missing and the car engine volume is changing weirdly depending on the RPM. Feels like 5.1 or 7.1 is turned on on a 2 speakers or on headset.

log:
[steam-1462810.log](https://github.com/ValveSoftware/Proton/files/7402783/steam-1462810.log)

could be problem on my system also though, but haven't noticed that issue on any other game as of yet.

weird parts in the log:
```
57112.961:0140:0144:warn:debugstr:OutputDebugStringA "D:\\Dev\\PlayAll\\Code\\Runtime\\Common\\PlayAll\\Common\\Core\\MountTool.cpp(121) : warning : Mount failed : Path = <COMMON/SOUNDS/VOICES/EN>  Virtual Drive = <VOICES:/>\n"
57112.961:0140:0144:trace:seh:dispatch_exception code=40010006 flags=0 addr=000000007B01236E ip=000000007B01236E tid=0144
57112.961:0140:0144:trace:seh:dispatch_exception  info[0]=00000000000000a3
57112.961:0140:0144:trace:seh:dispatch_exception  info[1]=000000000063a5e0
57112.961:0140:0144:warn:seh:dispatch_exception "D:\\Dev\\PlayAll\\Code\\Runtime\\Common\\PlayAll\\Common\\Core\\MountTool.cpp(121) : warning : Mount failed : Path = <COMMON/SOUNDS/VOICES/EN>  Virtual Drive = <VOICES:/>\n"
```

edit: seems there are people on windows also having this issue: https://steamcommunity.com/app/1462810/discussions/0/3085520348624868994/

**Comment by xpander69** [comment]
Ok fixed the audio issue by editing:
`~/SteamLibrary/steamapps/compatdata/1462810/pfx/drive_c/users/steamuser/Documents/My Games/WRC10/UserSettings.cfg`

and setting spatial audio to false 

`Engine.Audio.SpatialMix = false;`

seems to affect windows users also, so probably not a proton bug

**Comment by xpander69** [comment]
Ok one more weird issue i noticed. Im not sure if its the game itself (i don't have windows to try) or something in proton.
Basically i have G920 and a cheap chinese Arduino based Handbreak. Everything is nicely detected in the game, including handbreak axis. But the problem is that when i hit the handbreak i will lose the output from my wheel when 1 of the axis is active on my wheel. so i have to release lets say acceleration for a sec and then i can use it again. Its like both of those devices aren't used simultaneously. its either one or other. 
Not sure how to debug this or if anyone on windows can try this. Same thing seems to happen when i map the handbreak to my keyboard button for example. Might be a game bug also ofc.

**Comment by TheAquabat** [comment]
the fix proposed doesn't solve the issue with my controller I have a GameSir T4 pro, it can work as a Xbox One controller or an Xbox 360 controller none of them work.

**Comment by TheAquabat** [comment]
the fix proposed doesn't solve the issue with my controller I have a GameSir T4 pro, it can work as a Xbox One controller or an Xbox 360 controller none of them work.
(I tried the bleeding edge branch of proton experimental.)

here's a log with Proton experimental 

https://drive.google.com/file/d/1jbKIZKA6BlS2ZaKLQbd6K34IBqDDVj8D/view?usp=sharing

but also I must mention that a previous steam beta broke controller support on all games for me, it was fixed with another beta update, but maybe it wasn't completely fixed.
and also it's strange because the start button will work on the start screen, but if I go to the configuration menu there is no controller to select.


EDIT: it works but setting on big picture to use the controller to control the desktop, like a virtual mouse and keyboard... so to speak in "emulation mode" then I have to configure the key bindings one by one, which is not ideal.

EDIT2: I made it work but with Proton 6.15-GE-2, I first updated vk3d, then ran **protontricks 1462810 winecfg** set windows to windows 7, and then added this to the command launch options **WINEDLLOVERRIDES=”xinput1_3=b” %command%**. (also remember to delete/recreate the pfx prefix, first)
Seems to me that something has regressed on proton experimental (and other proton version like proton ge 6.20) that makes the controller not work even using basically the same workarounds.

**Comment by ivyl** [comment]
We [had to](https://github.com/ValveSoftware/Proton/commit/cc9c109bf6f606e00aa732607f65b523de5b21c0) switch to [`STEAM_COMPAT_CONFIG`](https://github.com/ValveSoftware/Proton#runtime-config-options), so instead of the override it may be necessary to `STEAM_COMPAT_CONFIG="" %command%` in addition to using win7 prefix.

**Comment by TheAquabat** [comment]
yes can confirm that command works in WRC8, which esentially has the same issue. (I refunded WRC10 tho)
thanks

**Comment by Samsagax** [comment]
> We [had to](https://github.com/ValveSoftware/Proton/commit/cc9c109bf6f606e00aa732607f65b523de5b21c0) switch to [`STEAM_COMPAT_CONFIG`](https://github.com/ValveSoftware/Proton#runtime-config-options), so instead of the override it may be necessary to `STEAM_COMPAT_CONFIG="" %command%` in addition to using win7 prefix.

I'm using Big Picture Mode with controller support for a PS4, I've installed WRC 10 yesterday and worked out of the box with proton 6.3-8 (current one). But the controller I'm using is not recognized, only mouse and keyboard (not very playable).
Tried the suggested fix (setting `STEAM_COMPAT_CONFIG="" %command%` as launch option) but didn't work, the game showed a crash report utility.
Tested other proton versions (GE-6.20, GE-7.1) with and without the launch option and had the same result except this time the crash report utility didn't show up.

Does setting the prefix to win7 made any difference in this? I'm not sure how to configure it.

**Comment by Samsagax** [comment]
New tests I made:
Proton 7 as is won't recognize the PS4 controller.
I tested proton 6.3-8 with the Windows version set to "win7" and the PS4 controller works without any launch option configuration. But the buttons are scrambled: Y axis on sticks are reversed, A and B buttons are Circle and Square. Triggers are scrambled too, mapping to "right stick left".
Then setting `WINEDLLOVERRIDES=”xinput1_3=b” %command%` as command line option yielded the same result. The controller is recognized, but the mappings are scrambled.

After the fix @ivyl suggested (setting `STEAM_COMPAT_CONFIG="" %command%` as launch option) The game did not start at all showing an error report window under proton 6.3-8. Under proton 7.0-1 after setting the windows version (via registry editing or winecfg) the game starts and the PS controller is recognized with it's own correct mapping.

So maybe it is worth removing [this change](https://github.com/ValveSoftware/Proton/commit/cc9c109bf6f606e00aa732607f65b523de5b21c0) for this title and also override the Windows version as "win7" in the user registry.

**Comment by BruebachL** [comment]
I'm unable to either join or create a lobby in any kind of multiplayer menu. Anyone experiencing a similar issue?

**Comment by pollux78** [comment]
In 2024 this game works perfectly under proton 9, no audio issues, played a public multiplayer match, my 8bitdo controller is working properly, no fps problems on my amd card either, works better under dxvk tho

Would like to see if this game can be changed to playable as i dont see any major problems anymore

This game is still working great for me close to the end of 2024, detects my wheel some feedback issues like it being delayed but i can use ffbtools to fix that

**Comment by xpander69** [comment]
How much are you throttling? does ffbtools actually work for you? I don't have WRC 10 installed atm, but WRC Generations spams the commands also and throttling time 16-24 range fixes all that for my G920 at least.

**Comment by pollux78** [comment]
> How much are you throttling? does ffbtools actually work for you? I don't have WRC 10 installed atm, but WRC Generations spams the commands also and throttling time 16-24 range fixes all that for my G920 at least.

Yes without throttling it i get a lot of delay with the wheel, with throttling its just that kernel spamming that happens, not that big of a issue

I was using around 22 to throttle it 

/home/pollux/ffbtools/bin/ffbwrap --throttling --throttling-time 22 /dev/input/by-id/usb-Logitech_G920_Driving_Force_Racing_Wheel_for_Xbox_One_000092fa4b58ed30-event-joystick

EDIT: im a dumbo i dont have %command% in it :P

**Comment by xpander69** [comment]
AFAIK on the logitech drivers on windows do that throttling by default and its 9ms there iirc. But yeah below 16ms for me causes the command queue getting full. Most of it is just the stage start when it initializes all the effects. 

Hehe. So your command wasnt actually doing anything :)

you need -- also afaik
so 
`/home/pollux/ffbtools/bin/ffbwrap --throttling --throttling-time 22 /dev/input/by-id/usb-Logitech_G920_Driving_Force_Racing_Wheel_for_Xbox_One_000092fa4b58ed30-event-joystick -- %command%`

**Comment by pollux78** [comment]
> AFAIK on the logitech drivers on windows do that throttling by default and its 9ms there iirc. But yeah below 16ms for me causes the command queue getting full. Most of it is just the stage start when it initializes all the effects.
> 
> Hehe. So your command wasnt actually doing anything :)
> 
> you need -- also afaik so `/home/pollux/ffbtools/bin/ffbwrap --throttling --throttling-time 22 /dev/input/by-id/usb-Logitech_G920_Driving_Force_Racing_Wheel_for_Xbox_One_000092fa4b58ed30-event-joystick -- %command%`

one issue im seeing is with the throttle the wheel does come up with failed to send command to device requiring me to replug it back in to make the ffb work in the game until i restart it again

**Comment by xpander69** [comment]
> 
> one issue im seeing is with the throttle the wheel does come up with failed to send command to device requiring me to replug it back in to make the ffb work in the game until i restart it again


Hmm. I don't have such issue.

`WINEDLLOVERRIDES="xinput1_3=n" ENABLE_VKBASALT=1 mangohud ffbwrap --throttling --throttling-time 19 /dev/input/by-id/usb-Logitech_G920_Driving_Force_Racing_Wheel_for_Xbox_One_0000c58d7f6b6a4c-event-joystick -- %command%`

with that i don't see any kernel messages on dmesg about command queue and FFB effects work great on game, no delays or issues.

edit:  
thats the log for me with no command queue failures.
 https://pastebin.com/jtBsdpfX

**Comment by John-Development** [comment]
Having issues making the game detect the FFB of my Thrustmaster T248. Everything else works fine. Using the [hid-tmff2](https://github.com/Kimplul/hid-tmff2) kernel patch with the thrustmaster driver installed via protontricks.

Tweaking through the config of WRC10 I found that although it is applying a Thrustmaster config, is not able to recognize that the controller has FFB, and the FFB toggle is disabled (tooltip says something like: This option is only available for controllers with force feedback).

In windows, the same configuration works fine.
